workflows/prognostic_c48_run/runtime/factories.py

```diff
@@ -105,17 +101,22 @@ def get_tendency_prescriber(
 
 def _get_time_lookup_function(
     mapper: Mapping[str, xr.Dataset],
-    variables: Sequence[str],
+    variables: Union[Sequence[str], Mapping[str, str]],
     initial_time: Optional[str] = None,
     frequency_seconds: float = 900.0,
     limiter: Optional[DatasetQuantileLimiter] = None,
 ) -> Callable[[cftime.DatetimeJulian], State]:
+    if isinstance(variables, list):
+        _variables: Mapping[str, str] = {var: var for var in variables}
+    elif isinstance(variables, dict):
+        _variables = variables
```

Comment:
Better to have behavior not depend on argument type. See other comment on `PrescriberConfig`.